scripts/weekly-dep-report.sh

```diff
@@ -0,0 +1,87 @@
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind thirdparty --omit-edges-into diem-workspace-hack | wc -l | tr -d ' ')
+}
+
+get_total_direct_deps() {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind directthirdparty --omit-edges-into diem-workspace-hack | wc -l | tr -d ' ')
+}
+
+get_total_dsr_deps () {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind thirdparty --omit-edges-into diem-workspace-hack -p safety-rules | wc -l | tr -d ' ')
+}
+
+get_total_direct_dsr_deps () {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind directthirdparty --omit-edges-into diem-workspace-hack -p safety-rules | wc -l | tr -d ' ')
+}
+
+get_total_dups () {
+    echo $(cargo +stable guppy dups --target x86_64-unknown-linux-gnu | wc -l | tr -d ' ')
+}
+
+
+git checkout --quiet "${last_commit}"
+date=$(git show --no-patch --format=%cd --date='format:%Y-%m-%d' ${rev})
+short_sha=$(git rev-parse --short HEAD)
+total=$(get_total_deps)
+direct=$(get_total_direct_deps)
+lsr_total=$(get_total_dsr_deps)
+lsr_direct=$(get_total_direct_dsr_deps)
+dups=$(get_total_dups)
+
+if [[ -z "${first_commit_within_last_seven_days}" ]]; then
+    #case 1: no commit made in last seven days
+    :
+elif [[ "${last_commit}" == "${first_commit_within_last_seven_days}" ]]; then
+    # case 2: only one commit made in last seven days
+    second_last_commit=$(git log --skip=1 --pretty=format:'%H' | head -n 1)
+    git checkout --quiet "${second_last_commit}"
+else
+    # case 3: more than one commit made in last seven days
+    last_commit_from_last_week=$(git log --pretty=format:'%H' "${first_commit_within_last_seven_days}"^1 | head -n 1)
+    git checkout --quiet "${last_commit_from_last_week}"
+fi
+
+prev_date=$(git show --no-patch --format=%cd --date='format:%Y-%m-%d' ${rev})
+prev_short_sha=$(git rev-parse --short HEAD)
+prev_total=$(get_total_deps)
+prev_direct=$(get_total_direct_deps)
+prev_lsr_total=$(get_total_dsr_deps)
+prev_lsr_direct=$(get_total_direct_dsr_deps)
+prev_dups=$(get_total_dups)
+
+change_total=$(format_change_in_dependency_output $((total-prev_total)))
+change_direct=$(format_change_in_dependency_output $((direct-prev_direct)))
+change_lsr_total=$(format_change_in_dependency_output $((lsr_total-prev_lsr_total)))
+change_lsr_direct=$(format_change_in_dependency_output $((lsr_direct-prev_lsr_direct)))
+change_dups=$(format_change_in_dependency_output $((dups-prev_dups)))
+
+#resetting HEAD to starting point
+git checkout --quiet "${main_branch}"
```

Comment:
```suggestion
git checkout --quiet "${last_commit}"
```
Since if its HEAD, then you don't actually go back to the starting point :)

Reply:
oh okay. Yeah, that was a silly bug. I was assuming parameter passed as "main"